File: EngineerProject.Mobile/Services/HomeService.cs
using EngineerProject.Commons.Dtos;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace EngineerProject.Mobile.Services
{
    public class HomeService : BaseService
    {
        public async Task<RequestResponse> AskForInvite(int id)
        {
            var response = await client.PostAsync($"Groups/AskForInvite", new { id });

            return MapResponse(response);
        }

        public async Task<RequestResponse> Create(string name, string description, bool isPrivate)
        {
            var response = await client.PostAsync($"Groups/Create", new { name, description, isPrivate });

            return MapResponse(response);
        }

        public async Task<DataRequestResponse<List<GroupTileDto>>> GetGroups(int page, int pageSize, string filter)
        {
            var response = await client.GetAsync<List<GroupTileDto>>($"Groups/Get?page={page}&pageSize={pageSize}&filter={filter}");

            return MapResponse(response);
        }

        public async Task<DataRequestResponse<UserGroupsWrapperDto>> GetUserGroups()
        {
            var response = await client.GetAsync<UserGroupsWrapperDto>($"Groups/GetUserGroups");

            return MapResponse(response);
        }

        public async Task<RequestResponse> ResolveGroupInvite(int id, bool value)
        {
            var response = await client.PostAsync($"Groups/ResolveGroupInvite", new { id, value });

            return MapResponse(response);
        }
    }
}

File: EngineerProject.Web/src/app/home/services/home.service.ts
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { GroupTile } from 'src/app/home/interfaces/GroupTile';
import { HomeGroupWrapper } from '../interfaces/HomeGroupWrapper';
import { HttpRequestWrapperService } from 'src/app/core/services/http-request-wrapper.service';

@Injectable({
  providedIn: 'root'
})
export class HomeService {
  constructor(private httpRequestWrapper: HttpRequestWrapperService) {
  }

  getUserGroups() {
    const url = `${environment.apiUrl}groups/GetUserGroups`;

    return this.httpRequestWrapper.get<HomeGroupWrapper>(url);
  }

  getGroups(filter: string) {
    const url = `${environment.apiUrl}groups/Get?pageSize=5&page=1&filter=${filter}`;

    return this.httpRequestWrapper.get<GroupTile[]>(url);
  }

  createGroup(name: string, description: string, isPrivate: boolean) {
    const url = `${environment.apiUrl}groups/Create`;

    return this.httpRequestWrapper.post<GroupTile>(url, { name, description, isPrivate });
  }

  askForInvite(id: number) {
    const url = `${environment.apiUrl}groups/AskForInvite`;

    return this.httpRequestWrapper.postBoolean(url, { id });
  }

  resolveGroupInvite(id: number, value: boolean) {
    const url = `${environment.apiUrl}groups/ResolveGroupInvite`;

    return this.httpRequestWrapper.postBoolean(url, { id, value });
  }
}